pkg/controller/migmigration/restore.go

```diff
@@ -232,6 +233,41 @@ func (t *Task) updateRestore(restore *velero.Restore, backupName string) {
 	t.updateNamespaceMapping(restore)
 }
 
+func (t *Task) cleanRestores() error {
```

Comment:
Seems like this could be simpler using the correlation tags.
Instead of deleting specific restores, how about listing all restores created by this migration and delete them (in a loop) using the correlation labels?